### ISSUE TITLE
[Question]: Suddenly slow indexing

### ISSUE THREAD

**Issue by a3lita** [run-llama/llama_index#10379]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

Hi, I'm working on a RAG application to parse complex PDFs and hosting it on Streamlit. Up until recently, indexing took maybe 10-15 seconds. Now, it's taking minutes (and there are a lot more progress bars) and the derived responses seem to be lower quality. I did not change my indexing function at all, so I'm at a loss for what caused this behavior. Below is my indexing function.
```
def get_index(documents):
    client = qdrant_client.QdrantClient(location=":memory:")
    vector_store = QdrantVectorStore(client=client, collection_name="test_store")

    # create the ingestion pipeline with transformations
    pipeline = IngestionPipeline(
        transformations=[
            SentenceSplitter(chunk_size=200, chunk_overlap=10), # adjust chunk size and overlap
            TitleExtractor(), # metadata extraction (extracts title)
            OpenAIEmbedding(model='text-embedding-ada-002'), # embeddings are calculated as part of the pipeline
        ],
        vector_store=vector_store, # set vector store to qdrant store
    )

    # run the pipeline to get nodes
    nodes = pipeline.run(documents=documents)

    # create vector store index
    from llama_index import VectorStoreIndex
    index = VectorStoreIndex.from_vector_store(vector_store)

    return index
```

Any suggestions would be much appreciated, thanks!

**Comment by logan-markewich** [comment]
Your code seems fine to me.

As the bot was kind of suggesting, did your data change? If you want, you can also set `pipeline.run(..., num_workers=4)` or similar to enable multiprocessing

**Comment by a3lita** [comment]
Nope, the data didn't change at all. Interestingly, the indexing actually still seems to work fine when the app is hosted on Streamlit community cloud, but when I run the Streamlit app locally, the slowdown happens (it used to work fine both ways). I pulled the same code from my repo as is being used for the hosted app to make sure there weren't any discrepancies, so I'm pretty confused on what's suddenly causing this

**Comment by a3lita** [comment]
@dosu-bot thank you for the suggestions. I am actually privately hosting this app on Streamlit community cloud, and indexing still takes a reasonable amount of time there (~45 seconds) whereas it suddenly takes a good couple of minutes+ locally (and shows a lot more progress bars than it used to). The code hasn't changed at all (I also tested this with an earlier version of my code that ran fine locally before), so I'm confused as to what could've caused this. I've tried multiprocessing during my ingestion pipeline to speed things up, but it still doesn't index nearly as fast as it used to. The data I'm using has not changed at all, either. I'm assuming the issue is with my machine or perhaps dependencies, but any ideas would be much appreciated, thanks!
